server/tests/common/backend.py

```diff
@@ -68,6 +69,35 @@ def app(backend: Backend, monkeypatch: pytest.MonkeyPatch) -> Iterator[AsgiApp]:
     asgi_app.state.backend = backend
     yield asgi_app
 
+@contextmanager
+def run_asgi_server(app: AsgiApp) -> Iterator[tuple[str, int]]:
+    from multiprocessing import Process
```

Comment:
Add this comment to explain why we have to rely on multiprocessing:
```
# Uvicorn is not designed to be run programmatically like we are going to do here
# (it contains calls to `sys.exit`, captures signal, and spawn tasks in the event loop).
# So instead we have to spawn it in it own sub-process...
#
# Note this comes with is own issues given it means the our test process gets
# forked and hence the `app`/`backend` objects diverge between the two processes
# (i.e. cannot send a request to the server then use the `backend` object to check
# how the server internal state has changed).
```

Reply:
done